x-pack/test/security_solution_api_integration/test_suites/investigation/saved_objects/tests/notes.ts

```diff
@@ -384,14 +384,15 @@ export default function ({ getService }: FtrProviderContextWithSpaces) {
 
       // skipped https://github.com/elastic/kibana/issues/196896
       describe('@skipInServerless', () => {
-        it('should retrieve all notes that have been created by a specific user', async () => {
+        // TODO we need to figure out how to retrieve the uid of the current user in the test environment
```

Comment:
@PhilippeOberti I'm afraid TODOs might stay here for long time so it's better to convert it to tech debts tickets for better visibility. Though you may try to solve it in this PR. I'm not sure what you mean by `uid`. Is it a user email or something else?

You might fetch the necessary data from the SAML Auth service. It also should help with new users creation. Something like below should help

```
const samlAuth = getService('samlAuth');

...

const userEmail = samlAuth.getEmail(); 
// or
const userData = samlAuth.getUserData(); 
```

If you think SAML Auth service doesn't provide enough functionality it might worth to ask @dmlemeshko.

Reply:
I am writing a ticket right now 😆 (great mind think alike!).

I need to be able to retrieve the entire UserProfile object. In the frontend we use the `uid` of the user to filter, as it seems that this is only guaranteed value to exist? I understand that we cannot create a user without a `username`, but the challenge here is that at the time of the note creation we use the `username` property [last](https://github.com/elastic/kibana/blob/main/packages/kbn-user-profile-components/src/user_profile.ts#L139) 🤦 and we cannot change that now as it would be a breaking change...
I was looking for a consistent way, and found this [bulkGet](https://github.com/elastic/kibana/blob/main/packages/core/user-profile/core-user-profile-server/src/service.ts#L43) method, which I'm now using in the `getNotes` endpoint. It searches users by `uid`...

Here's the user object that is returned to the frontend from the `getUserProfile` logic:
![Screenshot 2024-10-25 at 10 41 24 AM](https://github.com/user-attachments/assets/5a20acfd-d4a5-423b-8f10-27e17db5412d)

Does this make sense?

Reply:
I'm going to look into the saml auth service now, thank you!

Reply:
Yeah, got it. Technically SAML Provided should have user related information but I'm afraid not `uid`. It definitely looks like a task on it's own.

Reply:
I agree. I'm looking into it now, but if the `uid` isn't provided I'll keep the tests skipped and merge as is, as the scope of the PR would increase too much. This is an `8.16` bug fix 😆 
Thank you!

Reply:
logged the task [here](https://github.com/elastic/kibana/issues/197866) to be sure that we don't forget about it